### PR TITLE
ci: update dependabot config

### PR DESCRIPTION
**Pull request by csatib02**
## Overview

- Updated the dependabot config to group more dependencies together.
- Added label to the dependabot PR's.

### CODE REVIEW

**Review by sagikazarmark — APPROVED** [review]
LGTM

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -3,20 +3,29 @@ version: 2
...
       time: "16:00"
     open-pull-requests-limit: 10
     groups:
-      k8s:
+      k8s.io:
         patterns:
           - "k8s.io/api"
           - "k8s.io/apimachinery"
           - "k8s.io/client-go"
+          - "k8s.io/klog/v2"
```
Comment (sagikazarmark) [inline .github/dependabot.yaml:19]: wouldn't it make more sense to just use `k8s.io/*`?

Comment (csatib02) [reply .github/dependabot.yaml:19]: I did it this way, because I saw that previously it was used like this, but sure we can.

Comment (sagikazarmark) [reply .github/dependabot.yaml]: Maybe @ramizpolic can comment on the reasons

**Review by ramizpolic — APPROVED** [review]

**Pull request merged**